mender.go

```diff
@@ -539,9 +566,61 @@ func reauthorize(m *mender) func() (client.AuthToken, error) {
 	}
 }
 
+// setNextServer is an iterator like function that cycles through the
+// list of available servers in mender.menderConfig.Servers and puts
+// the server as "current active", i.e. fills out the fields
+// mender.menderConfig.ServerURL and mender.menderConfig.TenantToken
+// with the corresponding fields in the server list.
+func setNextServer(m *mender) func() string {
+	if m.config.Servers == nil {
+		return nil
+	}
+
+	var skipIdx int
+	idx := 0
+	numServers := len(m.config.Servers)
+	// find index to skip
+	for i := 0; i < numServers; i++ {
+		if m.config.ServerURL == m.config.Servers[i].ServerURL {
+			skipIdx = i
+			break
+		}
+	}
+	return func() (srvURL string) {
+		if m.config.Servers == nil {
+			return ""
+		}
+		if idx == skipIdx {
```

Comment:
So the returned function keeps the state of the variables of the outer function which defined it, even after both have returned? Interesting, didn't know this...